pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/SnapshotSegmentAbortedTxnProcessorImpl.java

```diff
@@ -0,0 +1,618 @@
+            WriteSegment,
+            DeleteSegment,
+            Close
+        }
+
+        private volatile OperationState operationState = OperationState.None;
+
+        ConcurrentLinkedDeque<Pair<OperationType, Supplier<CompletableFuture<Void>>>> taskQueue =
+                new ConcurrentLinkedDeque<>();
+        private CompletableFuture<Void> lastOperationFuture;
+        private final Timer timer;
+
+        public PersistentWorker(PersistentTopic topic) {
+            this.topic = topic;
+            this.timer = topic.getBrokerService().getPulsar().getTransactionTimer();
+            this.snapshotSegmentsWriterFuture =  this.topic.getBrokerService().getPulsar()
+                    .getTransactionBufferSnapshotServiceFactory()
+                    .getTxnBufferSnapshotSegmentService().createWriter(TopicName.get(topic.getName()));
+            this.snapshotIndexWriterFuture =  this.topic.getBrokerService().getPulsar()
+                    .getTransactionBufferSnapshotServiceFactory()
+                    .getTxnBufferSnapshotIndexService().createWriter(TopicName.get(topic.getName()));
+
+        }
+
+        public void appendTask(OperationType operationType, Supplier<CompletableFuture<Void>> task) {
+            switch (operationType) {
+                case UpdateIndex -> {
+                    if (!taskQueue.isEmpty()) {
+                        return;
+                    } else if (STATE_UPDATER.compareAndSet(this, OperationState.None, OperationState.UpdatingIndex)) {
+                        lastOperationFuture = task.get();
+                        lastOperationFuture.whenComplete((ignore, throwable) -> {
+                            if (throwable != null && log.isDebugEnabled()) {
+                                log.debug("[{}] Failed to update index snapshot", topic.getName(), throwable);
+                            }
+
+                            STATE_UPDATER.compareAndSet(this, OperationState.UpdatingIndex, OperationState.None);
+                        });
+                    }
+                }
+                case WriteSegment, DeleteSegment -> {
+                    taskQueue.add(new MutablePair<>(operationType, task));
+                    executeTask();
+                }
+                case Close -> {
+                    STATE_UPDATER.set(this, OperationState.Closing);
+                    taskQueue.clear();
+                    lastOperationFuture.thenRun(() -> {
+                        task.get().thenRun(() ->
+                                STATE_UPDATER.compareAndSet(this, OperationState.Closing, OperationState.Closed));
+                    });
+                }
+            }
+        }
+
+        private void executeTask() {
+            OperationType operationType = taskQueue.getFirst().getKey();
+            switch (operationType) {
+                case WriteSegment -> {
+                    if (STATE_UPDATER.compareAndSet(this, OperationState.None, OperationState.WritingSegment)) {
```

Comment:
now we have two thread to use this method, appendTask() and timer,  if one thread change the state to WritingSegment, and the second thread also can change state to WritingSegment when the task is finished. and the first thread removes the first task and change the state to `NONE`, the second task is DeletingSegment, how we change the state to `NONE`?